File: CS/RichEditEx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.Storage;
using Windows.UI.Xaml.Controls;

namespace ColosSchemesDemo1 {
    public class RichEditEx : RichEditBox {

        public RichEditEx() {
            DefaultStyleKey = typeof(RichEditEx);
        }

        protected Uri _Source;

        public Uri Source {
            get {
                return this._Source;
            }
            set {
                this._Source = value;
                this.OnSourceChanged();
            }
        }

        async void OnSourceChanged() {
            if (Source == null) {
                Document.SetText(Windows.UI.Text.TextSetOptions.None, null);
            }
            else {
                var file = await StorageFile.GetFileFromApplicationUriAsync(Source);
                var stream = await file.OpenAsync(Windows.Storage.FileAccessMode.Read);
                Document.LoadFromStream(Windows.UI.Text.TextSetOptions.FormatRtf, stream);
            }
        }
    }
}

File: VB/RichEditEx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports Windows.Storage
Imports Windows.UI.Xaml.Controls


Public Class RichEditEx
	Inherits RichEditBox

    Public Sub New()
	    DefaultStyleKey = GetType(RichEditEx)
    End Sub

	Protected _Source As Uri

	Public Property Source() As Uri
		Get
			Return Me._Source
		End Get
		Set(ByVal value As Uri)
			Me._Source = value
			Me.OnSourceChanged()
		End Set
	End Property

	Private async Sub OnSourceChanged()
		If Source Is Nothing Then
			Document.SetText(Windows.UI.Text.TextSetOptions.None, Nothing)
		Else
			Dim file = await StorageFile.GetFileFromApplicationUriAsync(Source)
			Dim stream = await file.OpenAsync(Windows.Storage.FileAccessMode.Read)
			Document.LoadFromStream(Windows.UI.Text.TextSetOptions.FormatRtf, stream)
		End If
	End Sub
End Class
